### ISSUE TITLE
Remove on a specific Entity and a subsequent find will retrieve the same entity with a Managed state

### ISSUE THREAD

**Issue by bertuz** [doctrine/orm#6123]
I don't know if it's the expected behavior, beh when removing a specific Entity, it is removed from the Identity Map. This implies that the subsequent finds within a single UoW will fetch the Entity with a 'MANAGED' state. Is this the expected behavior?

```

/**
     * INTERNAL:
     * Schedules an entity for deletion.
     *
     * @param object $entity
     *
     * @return void
     */
    public function scheduleForDelete($entity)
    {
        $oid = spl_object_hash($entity);

        if (isset($this->entityInsertions[$oid])) {
            if ($this->isInIdentityMap($entity)) {
                $this->removeFromIdentityMap($entity);
            }
```

**Comment by Ocramius** [comment]
@bertuz looks like a bug. Can you try reproducing it against https://github.com/doctrine/doctrine2/tree/d3f6c5ec70aac4b029a4b61ecf1e2ba61a1a4a6d/tests/Doctrine/Tests/ORM/Functional/Ticket please? (see that dir for examples)

**Comment by bertuz** [comment]
of cource, by tomorrow

**Comment by bertuz** [comment]
@Ocramius I was about to add a test into `Tests/ORM/Funcitonal/Ticket` but I don't know where to get the ticket number in order to name the file + open a pull req. Can you please show me the right way to that?

**Comment by Ocramius** [comment]
You can call it like this issue (6123)

**Comment by bertuz** [comment]
here you go https://github.com/doctrine/doctrine2/pull/6126

**Comment by xificurk** [comment]
Fixed by #11428
